BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/Entry.groovy

```diff
@@ -34,4 +34,29 @@ class Entry {
         value.call()
     }
 
+    Entry or(def fallback) {
+        def other = from(fallback)
+        new Entry(key, {
+            try {
+                return getValue()
+            } catch (Throwable e) {
+                try {
+                    return other.call()
+                } catch (Throwable e2) {
+                    throw CompositeException.from(e).add(e2)
```

Comment:
I was thinking if there were a way to, besides give a good description of the exceptions, nest the causes of the exceptions so it's clear from the stacktrace that if the fallback has failed was because the default value closure failed as well.

This is clear from both the spec and the exception message, but it could be even more explicit via the stacktrace.